scipy/special/xsf/zeta.h

```diff
@@ -1,10 +1,279 @@
+	-161.04003466346245,
+	-164.71578879628115,
+	-168.39154292909984,
+	-172.06729706191854,
+	-175.74305119473723,
+	-179.4188053275559,
+	-183.0945594603746
+    };
+
+    // Complex log of expansion coefficients for Euler-Maclaurin summation formula.
+    XSF_HOST_DEVICE inline std::complex<double> zeta_em_log_coeff(std::size_t n) {
+	std::complex<double> J(0.0, 1.0);
+	std::complex<double> result;
+	if (n < 50) {
+	    result = zeta_em_log_abs_coeff_lookup[n];
+	} else {
+	    // asymptotic formula
+	    result = std::log(2.0) - 2.0*(n + 1)*std::log(2*M_PI);
+	}
+	if (n % 2 == 1) {
+	    /* B_{2n}/(2n)! is negative for even n. This contributes a term
+	     * pi*i when taking the log. */
+	    result += M_PI * J;
+	}
+	return result;
+    }
+
+    /* Compute riemann_zeta for complex input z using the Euler-Maclaurin formula.
+     * Computation of individual terms in expansion are logarithmized to avoid
+     * overflow. TODO: only logarithmize when necessary. */
+    XSF_HOST_DEVICE inline std::complex<double> zeta_euler_maclaurin(std::complex<double> z) {
+	if (z.real() == 1.0) {
+	    return {std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN()};
+	}
+	std::size_t n = static_cast<std::size_t>(std::max(std::abs(z.imag()) / 4.0, 50.0));
+	std::size_t m = n;
+	std::complex<double> result = 0.0;
+	for (std::size_t i = 1; i < n; i++) {
+	    std::complex<double> term = std::pow(static_cast<double>(i), -z);
+	    result += term;
+	    if (std::abs(term) / std::abs(result) <= std::numeric_limits<double>::epsilon()) {
+		return result;
+	    }
+	}
+	double N = static_cast<double>(n);
+	std::complex<double> b = std::pow(n, -z);
+	result += b * (0.5 + N / (z - 1.0));
+	std::complex<double> log_poch = std::log(z);
+	std::complex<double> log_factor = std::log(b) - std::log(N);
+	for (std::size_t i = 0; i < m; i++) {
+	    std::complex<double> term = std::exp(zeta_em_log_coeff(i) + log_factor + log_poch);
+	    result += term;
+	    if (std::abs(term)/std::abs(result) <= std::numeric_limits<double>::epsilon()) {
+		return result;
+	    }
+	    log_poch += std::log(z + static_cast<double>(2*i + 1)) + std::log(z + static_cast<double>(2*i + 2));
+	    log_factor -= 2*std::log(N);
+	}
+	// Euler-maclaurin absolute error estimate.
+	double error;
+	error = std::abs(std::exp(zeta_em_log_coeff(m) + log_factor + log_poch));
+	error *= std::abs((z + 2.0*m + 1.0)/(z.real() + 2.0*m + 1.0));
+	// convert to relative error estimate
+	error /= std::abs(result);
+	if (error > 1e-8) {
+	    if (error > 1e-1) {
+		/* If error estimate predicts we don't even get 1 digit of precision, return NaN
+		 * and signal no result */
+		set_error("zeta", SF_ERROR_NO_RESULT, NULL);
+		return {std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN()};
+	    }
+	    // Signal reduced precision.
+	    set_error("zeta", SF_ERROR_LOSS, NULL);
+	}
+	return result;
+    }
+
+    constexpr double zeta_borwein_coeff[] = {
```

Comment:
Could you include the calculation script as a comment?

Reply:
It looks like these will be 

![image](https://github.com/user-attachments/assets/0a147282-5fb1-4c5f-9852-8b18f1bed7bb)

divided through by $d_n$, where $n = 50$?

Reply:
And would you mind commenting conceptually where these come from (Algorithm 2 of Borwein paper, $d_k$ divided through by $d_n$)?